### DISCUSSION TITLE
How to use mobx with solid.js?

### DISCUSSION THREAD

**Discussion by ghost** [mobxjs/mobx#3203] [Q&A]
Recently I have started watching videos on YouTube how to use mobx & i have seen 2 videos using mobx with react through mobx-react library & react Context, but have wonder how i use mobx with solid.js.
but, I can't find any library to integrate mobx with solid.js.
what you recommend?

**Comment by urugator** [comment]
I am not very familiar with `solidjs`, but I don't think there is much (if any) value in introducing `mobx` on the top of it. Just use it's [`createStore`](https://www.solidjs.com/docs/latest/api#createstore) or [`createMutable`](https://www.solidjs.com/docs/latest/api#createmutable).

**Reply by ghost** [reply]
yes! solid.js has has  builtin state managment mechanisms but i need to use more organized solution thats why we all use state management libraries. right!

**Reply by ghost** [reply]
solid js is really efficient becase it dont rerender component code, rather track dynamic values & update the ui & its production build is really small (about 5 or 6 kbs).

**Reply by urugator** [reply]
> more organized solution thats why we all use state management libraries. right!

Not sure what exactly do you mean by organized, but no, I don't think so. You don't need a library to organize state and mobx is actually quite unopinionated about organizing your state. Mobx is all about making your objects observable/trackable in a similar way as `solidjs` does.

**Reply by mweststrate** [reply]
IIRC mobx and solid are quite easy to integration as they are founded on the same principles, I thought Ryan had a PoC of that somewhere, but couldn't find it anymore. That being said, since solid already provides a very similar reactivity mechanism, I'm not sure if I would see the point of bringing MobX as well, I assume (never tried) Solid is similarly powerful in organising state

**Reply by ghost** [reply]
i know solid.js has a lot of state managment methods (hooks) like createSignal, createMemo, createMutable, createResource, more..., but what i need is to use mobx easyness in solid.js! last time i have used createResource with urql to query graphql server & store the result & it also has a refetch method, which is really cool!

**Reply by garkin** [reply]
SolidJS state management patterns doesn't work with big nested state as well as MobX for me.
Pretty tedious. Not Redux level of exhaustion, but still very laborious.
MobX in comparison is just effortless and straightforward.

**Reply by scott-cornwell** [reply]
@mweststrate @garkin I found this but it doesn't work right for me. It seems to have problems updating my nested state and just breaks randomly without error. You are underselling Mobx, it's a huge disappointment not to be able to use it with Solid. It's faster than React but just as tedious. Please make mobx-solid integration actually work!
import { Reaction } from 'mobx'
import { enableExternalSource } from 'solid-js'

// register MobX as an external source
let id = 0;
enableExternalSource((fn, trigger) => {
  const reaction = new Reaction(`mobx@${++id}`, trigger);
  return {
    track: (x) => {
      let next;
      reaction.track(() => (next = fn(x)));
      return next;
    },
    dispose: () => reaction.dispose()
  };
});

**Reply by scott-cornwell** [reply]
Ok actually the snippet I posted above does work fine with Solid! I was trying to use this CSS library called glory and I was having issues with that.

**Comment by gotjoshua** [comment]
Here is a fork from solids original sandbox with some more object types:
https://codesandbox.io/s/solid-mobx-with-array-obj-and-map-forked-updated-enhanced-zfql8d?file=/index.js

**Reply by js2me** [reply]
Hello currently I'm working on integration MobX with solid js.
https://js2me.github.io/mobx-solid/
